dataframe/series.go

```diff
@@ -0,0 +1,423 @@
+package dataframe
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"go.starlark.net/starlark"
+)
+
+// Series represents a sequence of values, either ints, floats, or objects. This is
+// the underlying data structure that is used to create DataFrames. A single column
+// of a DataFrame is a Series.
+type Series struct {
+	frozen    bool
+	which     int
+	valInts   []int
+	valFloats []float64
+	valObjs   []string
+	index     []string
+	dtype     string
+	name      string
+}
+
+// A Series contains values of one of these three types. The which field uses these
+// constants to determine which slice holds the actual values of the Series.
+const (
+	TypeInt   = 1
+	TypeFloat = 2
+	TypeObj   = 3
+)
+
+// Freeze prevents the series from being mutated
+func (s *Series) Freeze() {
+	s.frozen = true
+}
+
+// Hash cannot be used with Series
+func (s *Series) Hash() (uint32, error) {
+	return 0, fmt.Errorf("'Series' objects are mutable, thus they cannot be hashed")
+}
+
+// String returns the Series as a string in a readable, tabular form
+func (s *Series) String() string {
+	return s.stringify()
+}
+
+// Truth converts the series into a bool
+func (s *Series) Truth() starlark.Bool {
+	// NOTE: In python, calling bool(Series) raises this exception: "ValueError: The truth
+	// value of a Series is ambiguous. Use a.empty, a.bool(), a.item(), a.any() or a.all()."
+	// Since starlark does not have exceptions, just always return true.
+	return true
+}
+
+// Type returns the type as a string
+func (s *Series) Type() string {
+	return "dataframe.Series"
+}
+
+// Attr gets a value for a string attribute
+func (s *Series) Attr(name string) (starlark.Value, error) {
+	return nil, starlark.NoSuchAttrError(name)
+}
+
+// AttrNames lists available attributes
+func (s *Series) AttrNames() []string {
+	return []string{}
+}
+
+// Get retrieves a single cell from the Series
+func (s *Series) Get(keyVal starlark.Value) (value starlark.Value, found bool, err error) {
+	if name, ok := toStrMaybe(keyVal); ok {
+		pos := findKeyPos(name, s.index)
+		if pos == -1 {
+			return starlark.None, false, fmt.Errorf("not found: %q", name)
+		}
+		val, err := convertToStarlark(s.values()[pos])
+		if err != nil {
+			return starlark.None, false, err
+		}
+		return val, true, nil
+	}
+	if index, ok := toIntMaybe(keyVal); ok {
+		val, err := convertToStarlark(s.values()[index])
+		if err != nil {
+			return starlark.None, false, err
+		}
+		return val, true, nil
+	}
+	return starlark.None, false, fmt.Errorf("not found: %v", keyVal)
```

Comment:
`starlark.Value` includes a `String` method, so we should use `%q` or `%s` here instead of `%v`, which runs the risk of printing scary go stuff if you fed, say, a dict to `get`

Reply:
erm, this comment is still unresolved. Not the end of the world tho